contracts/Token.vy

```diff
@@ -69,14 +85,80 @@ def decimals() -> uint8:
 
 @external
 def transfer(receiver: address, amount: uint256) -> bool:
+    # Display msg.gas here to see what we start with.
+    log GasRemaining(msg.gas)
+
     assert receiver != ZERO_ADDRESS, "Cannot transfer to null address."
     self.balanceOf[msg.sender] -= amount
     self.balanceOf[receiver] += amount
 
     log Transfer(msg.sender, receiver, amount)
+
+    # Display msg.gas here to see what we end with.
+    log GasRemaining(msg.gas)
+
     return True
 
 
+struct Payment:
+    receiver: address
+    amount: uint256
+    
+
+@external
+def batchTransfer(payments: DynArray[Payment, MAX_PAYMENTS], min_gas_remaining: uint256 = MIN_GAS_REMAINING) -> uint256:
+    pay_count: uint256 = 0
+    pay_value: uint256 = 0
+    per_transfer_cost: uint256 = EST_GAS_PER_TRANSFER
+    gas_remaining : uint256 = msg.gas
+    gas_exhausted: bool = False
+
+    owner_balance : uint256 = self.balanceOf[msg.sender]
+
+    #log GasRemaining(msg.gas)
+
+    for payment in payments:
+
+        # Break if we don't have sufficient gas.
+        if msg.gas < (min_gas_remaining + per_transfer_cost): 
+            gas_exhausted = True
+            break
+
+        # We're complete if any receiver is a zero address.
+        if payment.receiver == ZERO_ADDRESS: break
+
+        # End if insufficient funds remaining during the batch.
+        if owner_balance < payment.amount: break
+
+        # If sender & receiver are different addresses then do the math.
+        if msg.sender != payment.receiver:
+            owner_balance -= payment.amount
+            self.balanceOf[payment.receiver] += payment.amount
+
+        #log GasRemaining(msg.gas)
+
+        log Transfer(msg.sender, payment.receiver, payment.amount)
```

Comment:
Here, for some reason, the `Transfer` event is not logged on polygonscan: https://mumbai.polygonscan.com/address/0xa5dfc266951e93f1cc2ed63be3dedf9a2d9ee507

Reply:
It is present in the `events` section.